ivy_tests/test_ivy/helpers.py

```diff
@@ -1901,3 +1901,53 @@ def num_positional_args(draw, *, fn_name: str = None):
     return draw(
         integers(min_value=num_positional_only, max_value=(total - num_keyword_only))
     )
+
+
+def bool_val_flags(cl_arg: Union[bool, None]):
+    if cl_arg is not None:
+        return st.booleans().filter(lambda x: x == cl_arg)
+    return array_bools(num_arrays=1)
+
+
+def handle_cmd_line_args(test_fn):
+    # first four arguments are all fixtures
+    def new_fn(get_command_line_flags, fw, device, call, *args, **kwargs):
```

Comment:
this design is wrong overall, we shouldn't overwrite anything unless the flag is set. We are currently overwriting with `array_bools(num_arrays=1)` if the flag is not set, but we should just leave the arguments unmodified if a flag has not been set.